pkgs/by-name/ic/icloudpd/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, nix-update-script
+, icloudpd
+, testers
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "icloudpd";
+  version = "1.17.3";
+  format = "pyproject";
+
+  src =  fetchFromGitHub {
+    owner = "icloud-photos-downloader";
+    repo = "icloud_photos_downloader";
+    rev = "v${version}";
+    sha256 = "sha256-GS6GqlZfj5kfjKLImkOTDAgQDGJQHl74uTqbZHVpbac=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    wheel
+    setuptools
+    requests
+    schema
+    click
+    python-dateutil
+    tqdm
+    piexif
+    urllib3
+    six
+    tzlocal
+    pytz
+    certifi
+    keyring
+    keyrings-alt
+  ];
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests = testers.testVersion { package = icloudpd; };
+  };
+
+  preBuild = ''
+    substituteInPlace pyproject.toml \
+      --replace "setuptools==69.0.2" "setuptools" \
```

Comment:
consider using pythonRelaxDepsHook

Reply:
I tried using it, but I don't think it works as the specific versions of `setuptools` and `wheel` are specified as in build system section

https://github.com/icloud-photos-downloader/icloud_photos_downloader/blob/master/pyproject.toml#L1-L5